e2e/rbd.go

```diff
@@ -55,104 +55,103 @@ var (
 	defaultCloneCount      = 10
 )
 
-func deployRBDPlugin() {
-	// delete objects deployed by rook
-	data, err := replaceNamespaceInTemplate(rbdDirPath + rbdProvisionerRBAC)
-	if err != nil {
-		e2elog.Failf("failed to read content from %s with error %v", rbdDirPath+rbdProvisionerRBAC, err)
-	}
-	_, err = framework.RunKubectlInput(cephCSINamespace, data, "--ignore-not-found=true", ns, "delete", "-f", "-")
-	if err != nil {
-		e2elog.Failf("failed to delete provisioner rbac %s with error %v", rbdDirPath+rbdProvisionerRBAC, err)
-	}
 
-	data, err = replaceNamespaceInTemplate(rbdDirPath + rbdNodePluginRBAC)
-	if err != nil {
-		e2elog.Failf("failed to read content from %s with error %v", rbdDirPath+rbdNodePluginRBAC, err)
-	}
-	_, err = framework.RunKubectlInput(cephCSINamespace, data, "delete", "--ignore-not-found=true", ns, "-f", "-")
-	if err != nil {
-		e2elog.Failf("failed to delete nodeplugin rbac %s with error %v", rbdDirPath+rbdNodePluginRBAC, err)
-	}
-
-	createORDeleteRbdResources("create")
+type Deployer interface {
+	replaceNamespaceInTemplate(string) (string,error)
+	runAction(string, string)
+	changeTemplateAndRunAction(string, string)
 }
 
-func deleteRBDPlugin() {
-	createORDeleteRbdResources("delete")
+// Keep the deployment templates in deployment order
+var rbdTemplates = []string{rbdDirPath + csiDriverObject,
```

Comment:
probably should place the 0th entry on a new line too, to make it easier to read